src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedEmbeddedAttributeSymbol.cs

```diff
@@ -27,29 +27,25 @@ internal class SynthesizedEmbeddedAttributeSymbol : NamedTypeSymbol
         private readonly string _name;
         private readonly NamedTypeSymbol _baseType;
         private readonly NamespaceSymbol _namespace;
-        private readonly ImmutableArray<MethodSymbol> _constructors;
         private readonly ModuleSymbol _module;
 
+        protected ImmutableArray<MethodSymbol> _constructors;
+
         public SynthesizedEmbeddedAttributeSymbol(
             AttributeDescription description,
             CSharpCompilation compilation,
-            Func<CSharpCompilation, NamedTypeSymbol, DiagnosticBag, ImmutableArray<MethodSymbol>> getConstructors,
-            DiagnosticBag diagnostics)
+            DiagnosticBag diagnostics,
+            bool generateDefaultConstructors = true)
         {
             _name = description.Name;
             _baseType = MakeBaseType(compilation, diagnostics);
 
-            if (getConstructors != null)
-            {
-                _constructors = getConstructors(compilation, this, diagnostics);
-            }
-            else
+            if (generateDefaultConstructors)
```

Comment:
I hate this. Any ideas on a better approach most welcome.

Reply:
Perhaps replace the `_constructors` field with an abstract `Constructors` property, and have two derived types, one that returns a single constructor with no parameters.

---
In reply to: [260050571](https://github.com/dotnet/roslyn/pull/33668#discussion_r260050571) [](ancestors = 260050571)